src/v/storage/segment.cc

```diff
@@ -870,24 +871,24 @@ ss::future<ss::lw_shared_ptr<segment>> make_segment(
             [path, pc, &resources, ntp_sanitizer_config](
               const ss::lw_shared_ptr<segment>& seg) mutable {
                 auto compacted_path = path.to_compacted_index();
-                return internal::make_compacted_index_writer(
-                         compacted_path,
-                         pc,
-                         resources,
-                         std::move(ntp_sanitizer_config))
-                  .then([seg, &resources](compacted_index_writer compact) {
-                      return ss::make_ready_future<ss::lw_shared_ptr<segment>>(
-                        ss::make_lw_shared<segment>(
-                          seg->offsets(),
-                          seg->release_segment_reader(),
-                          std::move(seg->index()),
-                          seg->release_appender(),
-                          std::move(compact),
-                          seg->has_cache()
-                            ? std::optional(std::move(seg->cache()->get()))
-                            : std::nullopt,
-                          resources));
-                  });
+                auto compact = make_file_backed_compacted_index(
+                  compacted_path,
+                  pc,
+                  false,
+                  resources,
+                  std::move(ntp_sanitizer_config));
+
+                return ss::make_ready_future<ss::lw_shared_ptr<segment>>(
```

Comment:
this particular diff hunk cleans up very nicely with the "ignore whitespace" option in github